cads_catalogue_api_service/client.py

```diff
@@ -98,6 +105,70 @@ def get_next_prev_links(
     return links
 
 
+def process_sanity_check(
```

Comment:
I would move `process_sanity_check` in a new file, so you can also add some pydantic types inside